### ISSUE TITLE
Cucumber messages issue: Attachments not displayed in the generated JSON when calling "attach" from hooks

### ISSUE THREAD

**Issue by Alexey-Stepanov** [cucumber/cucumber#1172]
## Summary

The issue occurs only when calling **attach** from hooks.rb
In this case, the generated cucumber-results.json does not contain screenshots after running cucumber-json-formatter

- Redundant "testStepFinished" section in NDJSON (just before the "attachment" field) when calling "attach" from hooks.rb
- Attachments are present when calling "attach" from step definition file

Please see the reproducible example below

## Expected Behavior

"embeddings" field is present in the JSON generated from cucumber messages file

## Current Behavior

No "embeddings" field in the JSON generated from cucumber messages file

## Steps to Reproduce

Example attached: [missing-screenshots-example.zip](https://github.com/cucumber/cucumber/files/5174907/missing-screenshots-example.zip)

## Your Environment

- Reproduced on MacOS 10.15.6 and Debian 10
- ruby-2.7.1
- cucumber 5.1.1
- [cucumber-json-formatter/v6.0.0](https://github.com/cucumber/cucumber/releases/tag/cucumber-json-formatter%2Fv6.0.0)

**Comment by Alexey-Stepanov** [comment]
This bug is confirmed and has a minimal reproducible example.
This is still very important to fix.

**Comment by clbond** [comment]
This is a critical issue for us as well. With `-f json` being deprecated it is very important to be able to attach inline screenshots, logs and videos to failed scenarios from hooks the way we always have before, without losing the data in the conversion process with https://github.com/cucumber/cucumber/tree/master/json-formatter

**Comment by ifosch** [comment]
We are also having this problem with the json formatter, keeping our reporting in Jenkins without screenshots, unless we use the deprecated JSON format. Is there anything we can do to help getting this fixed?

**Comment by luke-hill** [comment]
Hi @Alexey-Stepanov 

I've managed to reproduce this in one circumstance. However I've not had much time to look at it (And I know the deep-dive I've been wanting to do has been a bit hampered).

As a community we've re-prioritised this ticket onto a board so this will get looked at in the future. Sorry for any lack of information so far.

**Comment by aurelien-reeves** [comment]
@Alexey-Stepanov @clbond @ifosch
`json-formatter` has been fixed and released. v7.0.1 should now process attachments from hooks

**Comment by waynemd** [comment]
has anyone verified the fix for cucumber-json-formatter v7.0.1?  I still have issue after following steps here https://github.com/cucumber/cucumber/tree/master/json-formatter#cucumber-json-formatter to convert the attachment files ndjson file to json.  the attachment files on the cucumber report on jenkins is not displayable.  I see the size of the coverted json file is much smaller. am I missing something?

**Comment by aurelien-reeves** [comment]
@waynemd

Would you have the possibility to share an original ndjson file which would help us to try to reproduce the issue?

If so, don't forget to remove any sensitive data in it.

**Comment by waynemd** [comment]
@aurelien-reeves ,  can I share with you directly?

**Comment by aurelien-reeves** [comment]
Yes, this is possible
On the cucumber community slack for example? https://cucumberbdd.slack.com

**Comment by waynemd** [comment]
> Yes, this is possible
> On the cucumber community slack for example? https://cucumberbdd.slack.com

How to join this group?  It said my id "doesn’t have an account on this workspace."  Can I just send to your email directly?

**Comment by mattwynne** [comment]
@waynemd try https://cucumber.io/community#slack

**Comment by waynemd** [comment]
Thanks @mattwynne ,  I've been able to send the file to @aurelien-reeves directly in slack.

**Comment by aurelien-reeves** [comment]
@waynemd I look at your data.
Actually the files you tried to attach is not attached. The ndjson only contains the path of the file, not its content.

TL;DR: if you are using cucumber ruby, upgrade to v6.0. If not, please read the following.

Are you using cucumber ruby?
If so, we have just released a new version yesterday which should "fix" your issue.

Cf. cucumber ruby 6.0.0 [changelog](https://github.com/cucumber/cucumber-ruby/blob/master/CHANGELOG.md#v600) and the related issue: https://github.com/cucumber/cucumber-ruby/issues/1506

With cucumber ruby, before v6.0, you had to read the file your self and to give the content to the `attach` method.
From v6.0, you can now give it a path. A stream, a path, cucumber will take care of it.

If you are not using cucumber ruby but something else, you should look at the documentation for the `attach` method.

Which one are you using?
